web/src/stream_settings_ui.js

```diff
@@ -888,7 +888,7 @@ export function view_stream() {
     const row_data = get_row_data(active_data.$row);
     if (row_data) {
         const stream_narrow_hash =
-            "#narrow/stream/" + hash_util.encode_stream_name(row_data.object.name);
+            "#narrow/stream/" + hash_util.encode_stream_id(row_data.object.stream_id);
```

Comment:
I'm pretty sure that `stream_id` value is an `integer`; so `hash_util.encode_stream_id` should expect an integer (it only has two callers, and generally using integer IDs seems like this simpler/more natural API if we're tiebreaking).

You can test this code path by clicking the "preview channel" button in stream settings, I think.

Reply:
Ah yes, thanks! Changed `encode_stream_id` to take a number